x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/GeoPointScriptFieldType.java

```diff
@@ -0,0 +1,174 @@
+import org.elasticsearch.geometry.MultiPolygon;
+import org.elasticsearch.geometry.Point;
+import org.elasticsearch.geometry.Rectangle;
+import org.elasticsearch.index.mapper.GeoPointFieldMapper;
+import org.elasticsearch.index.mapper.GeoShapeQueryable;
+import org.elasticsearch.index.query.QueryShardContext;
+import org.elasticsearch.index.query.QueryShardException;
+import org.elasticsearch.script.Script;
+import org.elasticsearch.search.lookup.SearchLookup;
+import org.elasticsearch.xpack.runtimefields.fielddata.GeoPointScriptFieldData;
+import org.elasticsearch.xpack.runtimefields.query.GeoPointScriptFieldExistsQuery;
+import org.elasticsearch.xpack.runtimefields.query.GeoPointScriptFieldGeoShapeQuery;
+
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+public final class GeoPointScriptFieldType extends AbstractScriptFieldType<GeoPointFieldScript.LeafFactory> implements GeoShapeQueryable {
+
+    GeoPointScriptFieldType(String name, Script script, GeoPointFieldScript.Factory scriptFactory, Map<String, String> meta) {
+        super(name, script, scriptFactory::newFactory, meta);
+    }
+
+    @Override
+    protected String runtimeType() {
+        return GeoPointFieldMapper.CONTENT_TYPE;
+    }
+
+    @Override
+    protected Query rangeQuery(
+        Object lowerTerm,
+        Object upperTerm,
+        boolean includeLower,
+        boolean includeUpper,
+        ZoneId timeZone,
+        DateMathParser parser,
+        QueryShardContext context
+    ) {
+        throw new IllegalArgumentException("Field [" + name() + "] of type [" + typeName() + "] does not support range queries");
+    }
+
+    @Override
+    public Query termQuery(Object value, QueryShardContext context) {
+        throw new IllegalArgumentException(
+            "Geometry fields do not support exact searching, use dedicated geometry queries instead: [" + name() + "]"
+        );
+    }
+
+    @Override
+    public GeoPointScriptFieldData.Builder fielddataBuilder(String fullyQualifiedIndexName, Supplier<SearchLookup> searchLookup) {
+        return new GeoPointScriptFieldData.Builder(name(), leafFactory(searchLookup.get()));
+    }
+
+    @Override
+    public Query existsQuery(QueryShardContext context) {
+        checkAllowExpensiveQueries(context);
+        return new GeoPointScriptFieldExistsQuery(script, leafFactory(context), name());
+    }
+
+    @Override
+    public Query geoShapeQuery(Geometry shape, String fieldName, ShapeRelation relation, QueryShardContext context) {
+        if (shape == null) {
+            return new MatchNoDocsQuery();
+        }
+        final LatLonGeometry[] geometries = toLuceneGeometry(fieldName, context, shape);
+        return new GeoPointScriptFieldGeoShapeQuery(script, leafFactory(context), fieldName, geometries);
+    }
+
+    private static LatLonGeometry[] toLuceneGeometry(String name, QueryShardContext context, Geometry geometry) {
```

Comment:
Is this something we can move into core? It looks like something we'd have there already?

Reply:
Thats is my plan but it seems more difficult that it looks at first sight as every time we need to map a geometry to a Lucene geometry, it seems to be slightly different. As part of that effort I opened  https://issues.apache.org/jira/browse/LUCENE-9552 but I didn't have much support so far. I try to move it.